### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by Vynlar**
I'm just making this pull request in order to review your changes on this branch together instead of having to go search for them. Feel free to delete this or use it to merge into master.

### CODE REVIEW

File: src/components/App/index.js [hunk 1]
```diff
@@ -1,9 +1,54 @@
...
+        this.listener = this.props.firebase.auth.onAuthStateChanged(authUser => {
+            authUser
+                ? this.setState({ authUser })
+                : this.setState({ authUser: null });
+        });
+    }
+
+    componentWillUnmount()
+    {
+        this.listener();
```
Comment (Vynlar) [inline src/components/App/index.js:35]: Does calling the listener disable it? That seems like what you want here but I'm not familiar with the API.

File: src/components/Navigation/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const Navigation = ({ authUser} ) => (
+        <div>{authUser ? <NavigationAuth /> : <br />}</div>
+);
+
+const NavigationAuth = () => (
+    <div>
+        <h2>Navigation</h2>
+        <ul>
+            <li>
+                <Link to={ROUTES.ACCOUNT}>Account</Link>
```
Comment (Vynlar) [inline src/components/Navigation/index.js:18]: Using constants like this seems like a good idea. I think I'll steal this idea for the future

File: src/components/SignIn/index.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    <div>
+        <h1>SignIn</h1>
+        <SignInForm />
+    </div>
+)
+
+const INITIAL_STATE =
+{
+    email: '',
+    pswd: '',
```
Comment (Vynlar) [inline src/components/SignIn/index.js:18]:
```suggestion
    password: '',
```

File: src/components/SignIn/index.js [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    onChange = event =>
+    {
+        this.setState({ [event.target.name]: event.target.value });
+    };
+
+    render()
+    {
+        const { email, pswd, error } = this.state;
+    
+        const isInvalid = pswd === '' || email === '';
```
Comment (Vynlar) [inline src/components/SignIn/index.js:56]:
```suggestion
        const isInvalid = password=== '' || email === '';
```

File: src/components/SignIn/index.js [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            <form onSubmit={this.onSubmit}>
+                <input
+                    name="email"
+                    value={email}
+                    onChange={this.onChange}
+                    type="text"
+                    placeholder="Email Address"
+                />
+                <input
+                    name="pswd"
```
Comment (Vynlar) [inline src/components/SignIn/index.js:68]:
```suggestion
                    name="password"
```

File: src/components/SignIn/index.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+                    value={pswd}
```
Comment (Vynlar) [inline src/components/SignIn/index.js:69]:
```suggestion
                    value={password}
```

File: src/components/SignUp/index.js [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+class SignUpFormBase extends Component
+{
+    constructor(props) {
+        super(props);
+
+        this.state = {...INITIAL_STATE};
+    }
+
+    onSubmit = event => {
+        const { email, pswdA } = this.state;
```
Comment (Vynlar) [inline src/components/SignUp/index.js:33]:
```suggestion
        const { email, password } = this.state;
```

File: src/components/SignUp/index.js [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+    onChange = event => {
+        this.setState({ [event.target.name]: event.target.value });
+    };
+
+    render() {
+        const {
+            email,
+            pswdA,
+            pswdB,
```
Comment (Vynlar) [inline src/components/SignUp/index.js:56]:
```suggestion
            passwordConfirm,
```

**Review by Vynlar — COMMENTED** [review]
These are some suggested changes but structurally things look good!

File: src/components/Navigation/index.js (hunk 2 above)
Comment (lfryemason) [reply src/components/Navigation/index.js:18]: It seems to be a really easy way to keep track of the different routes.

**Pull request merged**